megamek/src/megamek/common/Terrain.java

```diff
@@ -389,6 +405,12 @@ public int movementCost(Entity e) {
                     && (moveMode == EntityMovementMode.INF_LEG) && (level > 1)) {
                 return level - 1;
             }
+            if (e.getCrew().getOptions().booleanOption(OptionsConstants.PILOT_TM_FOREST_RANGER)) {
+                if ((e instanceof Mech) && ((Mech)e).isSuperHeavy()) {
+                    return level - 2;
```

Comment:
Should this have a `Math.max(0, level -2)` instead?  I think this could go negative (although I didn't verify it too much).

Reply:
Went ahead and pushed a fix I had on the moveCost test branch I linked you in clack. This should work better than testing in each case.